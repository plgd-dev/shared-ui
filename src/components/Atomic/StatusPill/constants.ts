export const states = {
    ONLINE: 'online',
    OFFLINE: 'offline',
} as const
