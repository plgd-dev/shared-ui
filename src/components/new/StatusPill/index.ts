export { default } from './StatusPill'
export * from './StatusPill'
