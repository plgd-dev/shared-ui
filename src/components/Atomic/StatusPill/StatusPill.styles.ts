import { css } from '@emotion/react'
import { colors } from '../_utils/colors'
import { StatusPillStatusType } from './StatusPill.types'
import { states } from './constants'
import { ThemeType, get } from '../_theme'

export const statusPill = css`
    display: flex;
    align-items: center;
`

export const statusLine = (theme: ThemeType, status: StatusPillStatusType) => css`
    width: 4px;
    height: 24px;
    border-radius: 6px;
    background-color: ${status === states.ONLINE ? get(theme, 'StatusPill.statusLine.green') : get(theme, 'StatusPill.statusLine.red')};
`

export const label = css`
    display: inline-block;
    font-family: 'Poppins', sans-serif;
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    line-height: 14px;
    color: ${colors.neutral500};
`

export const content = css`
    padding-left: 8px;
`

export const pending = css`
    font-family: 'Poppins', sans-serif;
    font-style: normal;
    font-weight: 400;
    font-size: 10px;
    line-height: 12px;
    color: ${colors.primary};
    padding-top: 4px;
    cursor: pointer;
    white-space: nowrap;
`
