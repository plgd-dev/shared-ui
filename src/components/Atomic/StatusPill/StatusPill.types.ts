import { states } from './constants'
import { ReactNode } from 'react'

export type StatusPillStatusType = (typeof states)[keyof typeof states]

export type Props = {
    label: string
    status: StatusPillStatusType
    pending?: {
        text: string
        onClick: () => void
    }
    tooltipText?: ReactNode
}
