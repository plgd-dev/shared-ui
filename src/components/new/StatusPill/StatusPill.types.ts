import { states } from './constants'

export type StatusPillStatusType = typeof states[keyof typeof states]

export type Props = {
    label: string
    status: StatusPillStatusType
    pending?: {
        text: string
        onClick: () => void
    }
    tooltipText?: string
}
