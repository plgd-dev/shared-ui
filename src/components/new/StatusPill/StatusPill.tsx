import { FC } from 'react'
import { Props } from './StatusPill.types'
import * as styles from './StatusPill.styles'
import Tooltip from '../Tooltip'
import ConditionalWrapper from '../ConditionalWrapper'

const StatusPill: FC<Props> = (props) => {
    const { label, pending, status, tooltipText } = props
    return (
        <div css={styles.statusPill}>
            <div css={styles.statusLine(status)}></div>
            <div css={styles.content}>
                <ConditionalWrapper
                    condition={!!tooltipText}
                    wrapper={(child) => (
                        <Tooltip content={tooltipText} delay={200}>
                            {child}
                        </Tooltip>
                    )}
                >
                    <div css={styles.label}>{label}</div>
                </ConditionalWrapper>
                {pending && (
                    <div css={styles.pending} onClick={pending.onClick}>
                        {pending.text}
                    </div>
                )}
            </div>
        </div>
    )
}

StatusPill.displayName = 'StatusPill'

export default StatusPill
