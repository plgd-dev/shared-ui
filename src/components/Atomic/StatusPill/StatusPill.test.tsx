import React from 'react'
import { render } from '@testing-library/react'
import StatusPill from './StatusPill'

describe('<StatusPill>', () => {
    it('render correctly - snapshot', () => {
        const { asFragment } = render(
            <div>
                <StatusPill label='Online' status='online' tooltipText='Connected at: xxxxx' />
                <StatusPill
                    label='Offline'
                    status='offline'
                    tooltipText={
                        <span>
                            Last time online: <strong>31.9.2022 - 16:32</strong>
                        </span>
                    }
                />
                <StatusPill
                    label='Online'
                    pending={{
                        onClick: () => console.log('on click'),
                        text: '3 pending commands',
                    }}
                    status='online'
                    tooltipText='Connected at: xxxxx'
                />
                <StatusPill
                    label='Offline'
                    pending={{
                        onClick: () => console.log('on click'),
                        text: '3 pending commands',
                    }}
                    status='offline'
                    tooltipText={
                        <span>
                            Last time online: <strong>31.9.2022 - 16:32</strong>
                        </span>
                    }
                />
            </div>
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
