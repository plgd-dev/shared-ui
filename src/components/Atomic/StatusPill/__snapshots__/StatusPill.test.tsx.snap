// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusPill> render correctly - snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  width: 4px;
  height: 24px;
  border-radius: 6px;
  background-color: #52c5a2;
}

.emotion-2 {
  padding-left: 8px;
}

.emotion-3 {
  display: inline-block;
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 14px;
  color: #81868C;
}

.emotion-5 {
  width: 4px;
  height: 24px;
  border-radius: 6px;
  background-color: #D74E3A;
}

.emotion-12 {
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 10px;
  line-height: 12px;
  color: #2261AE;
  padding-top: 4px;
  cursor: pointer;
  white-space: nowrap;
}

<div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
      />
      <div
        class="emotion-2"
      >
        <div>
          <div
            class="emotion-3"
          >
            Online
          </div>
        </div>
      </div>
    </div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-5"
      />
      <div
        class="emotion-2"
      >
        <div>
          <div
            class="emotion-3"
          >
            Offline
          </div>
        </div>
      </div>
    </div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
      />
      <div
        class="emotion-2"
      >
        <div>
          <div
            class="emotion-3"
          >
            Online
          </div>
        </div>
        <div
          class="emotion-12"
        >
          3 pending commands
        </div>
      </div>
    </div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-5"
      />
      <div
        class="emotion-2"
      >
        <div>
          <div
            class="emotion-3"
          >
            Offline
          </div>
        </div>
        <div
          class="emotion-12"
        >
          3 pending commands
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
